src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,376 @@
+      setFocusedTime(date);
+    } else {
+      setFocusedTime(selectedTime ?? new Date());
+    }
+  };
+
+  const getHoursList = () => {
+    return use12Hours
+      ? [...new Array(12)].map((_, index) => (index === 0 ? 12 : index))
+      : [...new Array(24)].map((_, index) => index);
+  };
+
+  return (
+    <>
+      <TimePickerColumn
+        data={getHoursList()}
+        focusedTime={focusedTime}
+        selectedTime={selectedTime}
+        onFocusChange={onHourFocus}
+        onSelectChange={onHourClick}
+        isSame={(value, date) =>
+          isSameHour(value, date, use12Hours ? period : undefined)
+        }
+        step={hourStep}
+      />
+      {precision != 'hours' && (
+        <TimePickerColumn
+          data={[...new Array(60)].map((_, index) => index)}
+          focusedTime={focusedTime}
+          selectedTime={selectedTime}
+          onFocusChange={onMinuteFocus}
+          onSelectChange={onMinutesClick}
+          isSame={areSameMinutes}
+          step={minuteStep}
+        />
+      )}
+      {precision == 'seconds' && (
+        <TimePickerColumn
+          data={[...new Array(60)].map((_, index) => index)}
+          focusedTime={focusedTime}
+          selectedTime={selectedTime}
+          onFocusChange={onSecondFocus}
+          onSelectChange={onSecondsClick}
+          isSame={areSameSeconds}
+          step={secondStep}
+        />
+      )}
+      {use12Hours && (
+        <div className='iui-time'>
+          <ol>
+            {['AM', 'PM'].map((value) => (
+              <li
+                className={cx({
+                  'iui-selected': isSamePeriod(value, selectedTime),
+                })}
+                key={value}
+                tabIndex={isSamePeriod(value, selectedTime) ? 0 : undefined}
+                // ref={(ref) => {
+                //   scrollIntoView(ref, isSamePeriod(value, selectedTime));
+                // }}
+                onClick={() => onPeriodClick(value)}
+              >
+                {value}
+              </li>
+            ))}
+          </ol>
+        </div>
+      )}
+    </>
+  );
+};
+
+export type TimePickerColumnProps = {
+  /**
+   * Data to render in column.
+   */
+  data: number[];
+  /**
+   * Current focused time.
+   */
+  focusedTime: Date;
+  /**
+   * Current selected time.
+   */
+  selectedTime?: Date;
+  /**
+   * Callback when focus is changed.
+   */
+  onFocusChange?: (value?: number) => void;
+  /**
+   * Callback when date is changed.
+   */
+  onSelectChange?: (value: number) => void;
+  /**
+   * Compare function.
+   */
+  isSame: (value: number, date: Date | undefined) => boolean;
+  /**
+   * Step.
+   */
+  step?: number;
+};
+
+const TimePickerColumn = (props: TimePickerColumnProps): JSX.Element => {
+  const {
+    data,
+    focusedTime,
+    selectedTime,
+    onFocusChange,
+    onSelectChange,
+    isSame,
+    step = 0,
+  } = props;
+  const needFocus = React.useRef(false);
+
+  React.useEffect(() => {
```

Comment:
Add comment becaue we will forget again what it is for 😀

Reply:
added